### ISSUE TITLE
Please add a watch-list and progress tracker

### ISSUE THREAD

**Issue by kevwag** [Catch-up-TV-and-More/plugin.video.catchuptvandmore#849]
**Is your feature request related to a problem? Please describe.**
N/A

**Describe the solution you'd like**
What I'm proposing are two lists that a user could add shows to via the context menu. 

The first list would be a basic watch-list where episodes, movies or entire TV series can be added for viewing later on. 
This could possibly be in the form of a playlist.

The second list is a bit more complicated and would be a progress tracker / next episode list; 
The user would add a TV series from any of the addon's catchup branches (for example, Abandoned Engineering on UKTV Play), the list would be populated with the next unwatched episode from each show on the list. If the user was up-to-date with watching a show but new episodes air each week, the new episode would appear in the list. This gives the user a very useful, single place to access new content aggregated from every part of the plugin without having to browse through lots of different branches.

Last of all, it might be useful to link to one of Kodi's metadata scrapers (TVDB / TheMovieDB etc) for when there is no metadata available from the provideer. If the provider has metadata with the show, that would be used but if something is missing (Description / synopsis etc) then the external scraper could be used to find more data.
 
**Describe alternatives you've considered**
Trakt is a pretty good solution for keeping up with overall progress tracking but with the Kodi addon, it can only sync one-way. It can update the cloud progress with shows watched in the addon but it can't return a next-episode list to the addon. Even if it could do that, not all TV shows are available in this addon and Trakt has no way of filtering such a list.

**Additional context**
N/A

**Comment by joaopa00** [comment]
By design of CUTVM, it is impossible to do that. To fetch a streams informations are propagated along the files with actions of the user.

**Comment by darodi** [comment]
You could maybe use the "Add to add-on favourites" feature in the context menu instead.
(or "Add to favourites")

Your series/movies will appear in the Favourites section.
There you can also use the context menu and "Play from here"

Kodi already as a watched/unwatched status for each link. We should let it manage that part.
Trakt works both ways if I remember well... my kodi library was synced with the seen status from Trakt when I reinstalled Kodi.
It only works with the media collection though, not add-ons.

But then.... why don't we integrate CUTVM with kodi media library?
It's been done in other plugins! 

- https://github.com/add-ons/plugin.video.vrt.nu/pull/815
  - https://github.com/add-ons/plugin.video.vrt.nu/issues/833
- https://github.com/add-ons/plugin.video.vtm.go/wiki/Integratie-met-Kodi
  - in english https://github-com.translate.goog/add-ons/plugin.video.vtm.go/wiki/Integratie-met-Kodi?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=fr&_x_tr_pto=wapp


That would be a cool feature and Kodi's metadata scrapers would work too I think.

**Comment by kevwag** [comment]
> Kodi already has a watched/unwatched status for each link. We should let it manage that part.
Yes, it does but it doesn't have an aggregated list of your next unwatched episodes, which is what I was requesting.

Trakt's service IS capable of working both ways and it has been implemented in nearly every banned addon I've ever seen but that illustrates thee point I was making. The banned addons were designed to find and play pretty much anything so having a progress / up next list managed by Trakt made sense and worked really well. 

I agree, integrating with the Kodi library is definitely a step in the right direction and it would make the progress list easier to manage. Trakt integration into the addon would need to be added to complete the package: 

From what I remember of the banned addons, they created .strm files in their library folders so that if you played the content from the Kodi Library (outside of the addon), Kodi knew what plugin to use to play the stream. In our scenario we would use this the other way round - The .strm files could be used to filter the generic Trakt progress list and hide any shows that are not associated with the addon. Without the filter it would be a slow game of hit-or-miss, waiting for the addon to scrape every provider it supports for the content and maybe not finding it at all.
